lib/usb/usb.c

```diff
@@ -114,6 +115,11 @@ void usbd_register_resume_callback(usbd_device *usbd_dev,
 void usbd_register_sof_callback(usbd_device *usbd_dev, void (*callback)(void))
 {
 	usbd_dev->user_callback_sof = callback;
+
+	/* backend enable/disable */
+	if (usbd_dev->driver->enable_sof_callback) {
```

Comment:
is this really optional?  I kinda feel that if you're going to add support for registering a SOF callback, then it's either always available, or the feature is not available at all, without manually handling it.  Do you really need to check for this?  (maybe an assert... even though we have no asserts right now.....)